### PR TITLE
Simplified typings, for better readability and to enable proper tsdoc hints on create calls

### PR DESCRIPTION
**Pull request by ottokruse**
*Issue #, if available:* N/A

*Description of changes:* Simplified typings, for better readability and to enable proper tsdoc hints on create calls.

*Impact: This change will have no impact on regular usage of this library (e.g. all usage as explained in README). But, this might be a BREAKING change for TypeScript users, who were explicitly depending on the previous typings.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cognito-verifier.ts [hunk 1]
```diff
@@ -258,32 +275,19 @@ export class CognitoJwtVerifier<
...
-  ): CognitoIdOrAccessTokenPayload<
-    IssuerConfig,
-    { [key: string]: never } extends StillToProvide
-      ? OptionalProps
-      : MandatoryProps
-  > {
+  public verifySync<T extends SpecificVerifyProperties>(
+    ...args: { [key: string]: never } extends SpecificVerifyProperties
+      ? [jwt: string, props?: T & SpecificVerifyProperties]
+      : [jwt: string, props: T & SpecificVerifyProperties]
```
Comment (elorzafe) [inline src/cognito-verifier.ts:281]: Maybe that could be a type that could be reused on `verify` as well?

File: src/jwt-rsa.ts [hunk 2]
```diff
@@ -460,9 +475,9 @@ export abstract class JwtRsaVerifierBase<
    * @returns The payload of the JWT––if the JWT is valid, otherwise an error is thrown
    */
   public verifySync(
-    ...args: { [key: string]: never } extends StillToProvide
+    ...args: { [key: string]: never } extends SpecificVerifyProperties
       ? [jwt: string, props?: Partial<SpecificVerifyProperties>]
-      : [jwt: string, props: StillToProvide]
+      : [jwt: string, props: SpecificVerifyProperties]
```
Comment (elorzafe) [inline src/jwt-rsa.ts:480]: Can this be a type?

**Review by elorzafe — APPROVED** [review]
LGTM! well done @ottokruse 

Just one comment if is possible to have a type for the ones that are used on `verify` and `verifySync`?

**Comment by ottokruse** [comment]
Great idea, let me have a shot at it

**Review by elorzafe — APPROVED** [review]
LGTM! Good work @ottokruse 🎉

**Pull request merged**